ddl/db_integration_test.go

```diff
@@ -3162,6 +3162,163 @@ func (s *testIntegrationSuite3) TestDropTemporaryTable(c *C) {
+
+	tk.MustExec("drop global temporary table if exists temp")
+	tk.MustQuery("show warnings").Check(testkit.Rows())
+	err = tk.ExecToErr("select * from temp")
+	c.Assert(infoschema.ErrTableNotExists.Equal(err), IsTrue)
+}
+
+func (s *testIntegrationSuite3) TestDropLocalTemporaryTableWithDropTemporaryStmt(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("set tidb_enable_noop_functions=true")
+	tk.MustExec("set tidb_enable_global_temporary_table=true")
+	clearSQL := "drop table if exists tb, tb2, temp, temp1, ltemp1, ltemp2, testt.ltemp3"
+	tk.MustExec(clearSQL)
+	defer tk.MustExec(clearSQL)
+	// two normal table test.tb, test.tb2, a temporary table with name test.tb2
+	tk.MustExec("create table tb(id int)")
+	tk.MustExec("create table tb2(id int)")
+	tk.MustExec("insert into tb2 values(1)")
+	tk.MustExec("create temporary table tb2(id int)")
+	// two global temporary table test.temp, test.temp1
+	tk.MustExec("create global temporary table temp(id int) on commit delete rows")
+	tk.MustExec("create global temporary table temp1(id int) on commit delete rows")
+	// two local temporary table test.ltemp1, test.ltemp2
+	tk.MustExec("create temporary table ltemp1(id int)")
+	tk.MustExec("create temporary table ltemp2(id int)")
+	// a local temporary table test.ltemp3
+	tk.MustExec("create database if not exists testt")
+	tk.MustExec("create temporary table testt.ltemp3(id int)")
+
+	// testing for drop table which is not local temporary
+	err := tk.ExecToErr("drop temporary table tb")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table test.tb")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table temp1")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table test.temp1")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table ltemp1, tb")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table temp, ltemp1")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table xxx, ltemp1")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop temporary table xxx")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+
+	// testing for drop table if exists which is not local temporary
+	tk.MustExec("drop temporary table if exists xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.xxx'"))
+	tk.MustExec("drop temporary table if exists ltemp1, xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.xxx'"))
+	tk.MustExec("drop temporary table if exists tb1, xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.tb1,test.xxx'"))
+	tk.MustExec("drop temporary table if exists temp1, xxx")
```

Comment:
What if `drop temporary table if exists temp1`? We don't allow it.

Reply:
I'll add these case later

Reply:
Done